server/core/config_runtime.cc

```diff
@@ -175,6 +175,7 @@ bool runtime_link_server(Server* server, const char* target)
         {
             if (serviceAddBackend(service, server))
```

Comment:
The call to `service_update_weights` should be done inside `serviceAddBackend` and `serviceRemoveBackend`.